test/cppsuite/tests/bounded_cursor_stress.cpp

```diff
@@ -198,43 +199,40 @@ class cursor_bound_01 : public test {
      * bounds on the range cursor. The lower and upper bounds are randomly generated strings and the
      * inclusive configuration is also randomly set as well.
      */
-    std::pair<bound, bound>
-    set_random_bounds(thread_worker *tc, scoped_cursor &range_cursor)
+    bound_set
+    set_random_bounds(thread_worker *tc, scoped_cursor &bounded_cursor)
     {
         bound lower_bound, upper_bound;
 
-        testutil_check(range_cursor->reset(range_cursor.get()));
+        testutil_check(bounded_cursor->reset(bounded_cursor.get()));
         bound_action action =
           static_cast<bound_action>(random_generator::instance().generate_integer(0, 3));
         if (action == bound_action::NO_BOUNDS)
-            testutil_check(range_cursor->bound(range_cursor.get(), "action=clear"));
+            testutil_check(bounded_cursor->bound(bounded_cursor.get(), "action=clear"));
 
-        if (action == bound_action::LOWER_BOUND_SET || action == bound_action::ALL_BOUNDS_SET) {
+        if (action == bound_action::LOWER_BOUND_SET || action == bound_action::ALL_BOUNDS_SET)
             lower_bound = bound(tc->key_size, true);
-            range_cursor->set_key(range_cursor.get(), lower_bound.get_key().c_str());
-            testutil_check(
-              range_cursor->bound(range_cursor.get(), lower_bound.get_config().c_str()));
-        }
 
         if (action == bound_action::UPPER_BOUND_SET || action == bound_action::ALL_BOUNDS_SET) {
             if (action == bound_action::ALL_BOUNDS_SET) {
                 /* Ensure that the lower and upper bounds are never overlapping. */
-                if (_reverse_collator_enabled)
-                    upper_bound = bound(tc->key_size, false, lower_bound.get_key()[0] - 1);
-                else
-                    upper_bound = bound(tc->key_size, false, lower_bound.get_key()[0] + 1);
-            } else
+                auto diff = _reverse_collator_enabled ? -1 : 1;
+                upper_bound = bound(tc->key_size, false, lower_bound.get_key()[0] + diff);
+            } else {
                 upper_bound = bound(tc->key_size, false);
-            range_cursor->set_key(range_cursor.get(), upper_bound.get_key().c_str());
-            testutil_check(
-              range_cursor->bound(range_cursor.get(), upper_bound.get_config().c_str()));
+            }
         }
 
         if (action == bound_action::ALL_BOUNDS_SET)
             testutil_assert(
               custom_lexicographical_compare(lower_bound.get_key(), upper_bound.get_key(), false));
 
-        return std::make_pair(lower_bound, upper_bound);
+        if (!(lower_bound.get_key()).empty())
```

Comment:
np: i would remove the braces around `lower_bound.get_key()`, since its not needed

Reply:
[0270585](https://github.com/wiredtiger/wiredtiger/pull/8223/commits/02705857162292b83da6f37561cdbcce7636cbb8)